framework/Models/ExternalModel.py

```diff
@@ -128,8 +128,9 @@ def createNewInput(self,myInput,samplerType,**kwargs):
       newInput = ([(extCreateNewInput)],copy.deepcopy(kwargs))
     else:
       newInput =  Dummy.createNewInput(self, myInput,samplerType,**kwargs)
-    for key in kwargs['SampledVars'].keys():
-      modelVariableValues[key] = kwargs['SampledVars'][key]
+    if 'SampledVars' in kwargs.keys():
```

Comment:
This is because the SampledVars keyword was only searched for in the methods that were not previously called for preconditioner externalModels.  The original implementation suggests this conditional is appropriate, as it is not guaranteed to be present in other parts of this method.  However, we don't have a situation where this actually occurs until with the preconditioner, so it wasn't a problem.

Good catch!